### ISSUE TITLE
weirdness when setting config.vite.server.host

### ISSUE THREAD

**Issue by john-michael-murphy** [sveltejs/kit#4763]
### Describe the bug

When I set `config.vite.server.host` to something custom like: `localhost.avoid-cors-errors.com`, on first load, the dev server only spits out this: 

```
  SvelteKit v1.0.0-next.305

  local:   http://localhost:3000
  network: not exposed
  ```
  
  On second load, the dev server spits out this: 
  
  ```
  10:01:26 AM [vite] server restarted.

  > Local:    http://localhost.avoid-cors-errors.com:3000/
  > Network:  http://10.51.228.122:3000/
  ```
  
  The behavior I'd expect is to consistently see `local: http://localhost.avoid-cors-errors.com:3000/`, and to only see the server exposed to the network when passing the `--host` flag.

### Reproduction

Described above.

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 6.56 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.5.0 - ~/bin/nvm/versions/node/v17.5.0/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.4.1 - ~/bin/nvm/versions/node/v17.5.0/bin/npm
  Browsers:
    Chrome: 100.0.4896.127
    Firefox: 99.0
    Safari: 15.4
  npmPackages:
    @sveltejs/adapter-static: 1.0.0-next.29 => 1.0.0-next.29 
    @sveltejs/kit: 1.0.0-next.305 => 1.0.0-next.305 
    @sveltejs/vite-plugin-svelte: 1.0.0-next.41 => 1.0.0-next.41 
    svelte: ^3.44.0 => 3.47.0 
    vite: ^2.6.10 => 2.9.5
```

### Severity

annoyance

**Comment by john-michael-murphy** [comment]
Oh, if it's not already clear, this isn't at all urgent! And, as always, thanks so much for all the amazing work y'all are doing! Sveltekit is an amazing tool!

**Comment by bluwy** [comment]
I think this issue belongs in [Vite](https://github.com/vitejs/vite). I don't think there's anything SvelteKit can do here, unless it's only reproducible in SvelteKit?

**Comment by Conduitry** [comment]
The difference in the casing of "local" and "network" and the presence or lack of the ">" is peculiar. I don't know whether it's Vite or SvelteKit that's printing out each of these messages.

https://github.com/sveltejs/kit/blob/15516b460bc3c149b13a7c5282bef56feed1868c/packages/kit/src/cli.js is where it would be getting printed out if it were happening in SvelteKit presumably.

**Comment by john-michael-murphy** [comment]
@bluwy, I should have added this to original issue, but I haven't been able to reproduce this issue with a vanilla vite setup.

**Comment by bluwy** [comment]
Ah yeah with Conduitry's comment, it looks like the first time we're printing the message from SvelteKit, then when the server restarted (e.g. updating the svelte config), Vite prints its own message leading to the mismatch. I guess the second message from Vite needs to be turned off and log from SvelteKit instead.

**Comment by LucasSegersFabro** [comment]
hey @bluwy 
could we just log vite's `resolved_config` instead of `(server.httpServer).address()` (as per [proposed pr](https://github.com/sveltejs/kit/compare/master...LucasSegersFabro:gh-4763?expand=1#))?  

`address()` only returns localhost even if `vite.server.config.host` is set

Also, forked from `master` and vite (2.9.9) didn't double log

produced the following output
![image](https://user-images.githubusercontent.com/29109459/172082970-f70a9a99-6b64-4743-b7ad-d8a7e5efe7d5.png)

**Comment by bluwy** [comment]
Sorry this got lost in my notifications. I don't think that would work correctly as well because Vite still derives different URLs based on those properties, e.g. if `host` is `localhost`, it could show `127.0.0.1` or `[::1]` (at least for Vite 3). I think the longer term solution here is to have Vite expose the resolved URLs somewhere so we don't have to copy how Vite derives it, and would benefit other ecosystem that wraps the Vite dev server. I've made https://github.com/vitejs/vite/discussions/8891 for discussion.

Also with SvelteKit moving slowly as only a Vite plugin, we may be relying on the Vite dev server output instead, so this won't be an issue then.

**Comment by benmccann** [comment]
We no longer have the `svelte-kit` CLI, which means that this is either fixed (hopefully!) or should be filed as an issue in Vite instead

**Comment by john-michael-murphy** [comment]
Thanks so much, @benmccann

**Comment by hatemjaber** [comment]
I was looking for a way to use a "local domain" for testing purposes. I started a fresh new project and changed the following in vite.config
```
const config: UserConfig = {
	plugins: [sveltekit()],
	server: {
		host: "customdomain.com"
	}
};
```
I also started the server for extra measures with: `npm run dev -- --host customdomain.com`
Switching the server host and adding the flag to the start of the dev server prevents cookies from being set. Is there another place that needs to be notified of the server name change?
